lib/jxl/enc_modular.cc

```diff
@@ -1114,10 +1123,12 @@ Status ModularFrameEncoder::ComputeTree(ThreadPool* pool) {
 
     std::atomic_flag invalid_force_wp = ATOMIC_FLAG_INIT;
 
+    std::atomic<bool> has_error{false};
     std::vector<Tree> trees(useful_splits.size() - 1);
     JXL_RETURN_IF_ERROR(RunOnPool(
         pool, 0, useful_splits.size() - 1, ThreadPool::NoInit,
         [&](const uint32_t chunk, size_t /* thread */) {
```

Comment:
I feel like our code would be a lot simpler if RunOnPool bubbled up errors... WDYT?
In particular it would avoid having weird things in the else branch of ASSIGN_OR. (arguably, that code should always return anyway)

Reply:
Had the same idea. Though, let's do it in a separate PR.

Reply:
Agreed, but it might be worth it to do it in a separate PR that goes in *before* this one...

Reply:
Sure!